### PR TITLE
Handle Team ID from ASM PSTB

### PR DESCRIPTION
**Pull request by rapito**
Depends on https://github.com/kyhavlov/slippi-ssbm-asm/pull/2

### CODE REVIEW

File: Source/Core/Core/Slippi/SlippiNetplay.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ class SlippiPlayerSelections
 	u8 playerIdx = 0;
 	u8 characterId = 0;
 	u8 characterColor = 0;
+	u8 teamId = 0;
```
Comment (kyhavlov) [inline Source/Core/Core/Slippi/SlippiNetplay.h:47]: This needs to get added to the logic in `Merge()` below

File: Source/Core/Core/HW/EXI_DeviceSlippi.cpp [hunk 2]
```diff
@@ -2286,13 +2286,25 @@ void CEXISlippi::prepareOnlineMatchState()
 		}
 
 		// Overwrite local player character
-		onlineMatchBlock[0x60 + (lps.playerIdx-1) * 0x24] = lps.characterId;
+		onlineMatchBlock[0x60 + (lps.playerIdx - 1) * 0x24] = lps.characterId;
```
Comment (kyhavlov) [inline Source/Core/Core/HW/EXI_DeviceSlippi.cpp]: If you merge the doubles branch into yours, it should pull in the playerIdx changes so you don't need to subtract 1 here. I fixed it to be 0 indexed everywhere so it's not confusing anymore

File: Source/Core/Core/Slippi/SlippiNetplay.h (hunk 1 above)
Comment (rapito) [reply Source/Core/Core/Slippi/SlippiNetplay.h:47]: Done

File: Source/Core/Core/HW/EXI_DeviceSlippi.cpp (hunk 2 above)
Comment (rapito) [reply Source/Core/Core/HW/EXI_DeviceSlippi.cpp]: Done

**Pull request merged**